src/main/scala/com/streese/akka/Main.scala

```diff
@@ -38,6 +38,23 @@ object Main extends App {
       .via(ConnectionHandler.Actor.sinkAndSourceCoupledFlow(connectionHandler, 100, OverflowStrategy.fail))
       .map(res => ByteString(s"${res.n}\n"))
 
+    val connectionHandler2 =
+      spawnConnectionHandler2("tcp", connection.remoteAddress)
+
+    val flow2 = Flow[ByteString]
+      .via(Framing.delimiter(ByteString("\n"), maximumFrameLength = 256, allowTruncation = true))
+      .map(_.utf8String.stripSuffix("\r"))
```

Comment:
btw. if you could strip suffix before converting to string it would good for performance because no additional slightly truncated string will be created.